hydromt_sfincs/sfincs.py

```diff
@@ -560,12 +560,14 @@ def setup_gauges(self, gauges_fn=None, **kwargs):
         """
         if gauges_fn is not None:
             name = self._GEOMS["gauges"]
+            # ensure the catalog is loaded before adding any new entries
+            self.data_catalog.sources
             gdf = self.data_catalog.get_geodataframe(
-                filename, geom=self.region, assert_gtype="Point"
+                str(gauges_fn), geom=self.region, assert_gtype="Point", **kwargs
```

Comment:
good catch!

The data_catalog.get_geodataframe method now also handles Path instances, see https://github.com/Deltares/hydromt/pull/26